### PR TITLE
added more automation for data creation

### PR DESCRIPTION
**Pull request by donaldchen34**
Description of PR here...
-Added synonym checks for the nouns too
-The file now reads the sentences_to_train_on.txt to see what sentences to create training data on
-The file saves all unique sentences in unique_sentences.txt now
Issue: Closes #

### Frontend:
- 
### Backend:
- Updated data_creation.py

### CODE REVIEW

File: backend/machine learning/data_creation.py [hunk 1]
```diff
@@ -141,24 +158,40 @@ def save_Data():
...
 
 createFiles()
 
 data = [] #create_Data stores everything in data
-create_Data(action = action,noun = noun,color = color,sentence = sentence,
-            relative_action= relative_action,relative_object = relative_object,
-            relative_object_color =relative_object_color)
 
+
+with open(TRAINING_SENTENCES_FILE,'r') as csv_file:
```
Comment (JasperCheung) [inline backend/machine learning/data_creation.py]: can you write a main function and put this in it? `if __name__ == '__main__'`

File: backend/machine learning/data_creation.py [hunk 2]
```diff
@@ -141,24 +158,40 @@ def save_Data():
...
-relative_object_color = 'blue'
+    #Empty training_sentences file
+    with open(TRAINING_SENTENCES_FILE, 'w', newline='') as datafile:
+        csv.writer(datafile).writerow(FIELDS)
+        datafile.close()
 
 
 createFiles()
 
 data = [] #create_Data stores everything in data
```
Comment (JasperCheung) [inline backend/machine learning/data_creation.py]: move this down into the main function.

**Review by JasperCheung — CHANGES_REQUESTED** [review]
- Could you rename the `machine learning` dir to `machine_learning`?
- ~~running this doesn't do anything for me.~~
- It might make sense to .gitignore the `/data` folder. I don't think we wanna push a ton of sentences to the repo.
- add some comments homie
- add a readme

File: backend/machine_learning/data_creation.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+                         }
+                     }
+            json.dump(stats,outfile,indent=4)
+
+#Saves new sentence structures into sentence_structures.txt
+#Input: "Add the red cube" -> Sentence Structure: <target_action> the <color> <cube>
+#This helps to keep track of what sentence structures have been used so we do not repeat data
+#Returns false if the sentence structure is found in sentence_structures.txt
+#Returns true if not found and adds to sentence_structure.txt
+def save_Sentence_Structure(action,noun,color,sentence,relative_action=None,
```
Comment (donaldchen34) [inline backend/machine_learning/data_creation.py:184]: This is the new method of adding sentences:
Sentences will be added based on sentence structures so 
"Add the red cube near the green sphere" = "Add the green pyramid above the blue cube"
Both sentence structures are seen <Target_action> the <color> <noun> <rel_action> the <rel_obj_color> <rel_obj>
All unique combinations will be created based on the sentence structure

File: backend/machine_learning/data_creation.py [hunk 4]
```diff
@@ -82,14 +82,14 @@ def create_Data(action,noun,color,sentence,relative_action=None,
                     for rel_action_synoynm in RELATIVE_ACTIONS[relative_action.upper()]:
                         new_sentence2 = new_sentence1.replace(action, action_synonym)
                         new_sentence2 = new_sentence2.replace(relative_action,rel_action_synoynm)
-                        data.append([new_sentence2,action,noun_synonym,color,relative_action,
+                        data.append([new_sentence2,action,noun,color,relative_action,
```
Comment (donaldchen34) [inline backend/machine_learning/data_creation.py:85]: Data was using noun_synoynm as target. It should be noun

**Pull request closed without merging**